main.tf

```diff
@@ -247,3 +255,141 @@ resource "aws_s3_bucket_replication_configuration" "main" {
+          expired_object_delete_marker = try(expiration.value.expired_object_delete_marker, null)
+        }
+      }
+
+      dynamic "filter" {
+        for_each = [for v in try(flatten([rule.value.filter]), []) : v if max(length(keys(v)), length(try(rule.value.filter.tags, rule.value.filter.tag, []))) == 1]
+        content {
+          prefix                   = try(filter.value.prefix, null)
+          object_size_greater_than = try(filter.value.object_size_greater_than, null)
+          object_size_less_than    = try(filter.value.object_size_less_than, null)
+
+          dynamic "tag" {
+            for_each = try([filter.value.tag], [])
+
+            content {
+              key   = tag.value.key
+              value = tag.value.value
+            }
+          }
+        }
+      }
+
+      dynamic "filter" {
+        for_each = length(try(flatten([rule.value.filter]), [])) > 0 ? [] : [true]
+        content {
+        }
+      }
+
+      dynamic "filter" {
+        for_each = [for v in try(flatten([rule.value.filter]), []) : v if max(length(keys(v)), length(try(rule.value.filter.tags, rule.value.filter.tag, []))) > 1]
+        content {
+          and {
+            object_size_greater_than = try(rule.value.filter.and.object_size_greater_than, null)
+            object_size_less_than    = try(rule.value.filter.and.object_size_less_than, null)
+            prefix                   = try(rule.value.filter.and.prefix, "")
+            tags                     = try(rule.value.filter.and.tags, null)
+          }
+        }
+      }
+
+      dynamic "noncurrent_version_expiration" {
+        for_each = try(flatten([rule.value.noncurrent_version_expiration]), [])
+        content {
+          newer_noncurrent_versions = try(noncurrent_version_expiration.value.newer_noncurrent_versions, null)
+          noncurrent_days           = try(noncurrent_version_expiration.value.days, noncurrent_version_expiration.value.noncurrent_days, null)
+        }
+      }
+
+      dynamic "noncurrent_version_transition" {
+        for_each = try(flatten([rule.value.noncurrent_version_transition]), [])
+        content {
+          newer_noncurrent_versions = try(noncurrent_version_transition.value.newer_noncurrent_versions, null)
+          noncurrent_days           = try(noncurrent_version_transition.value.days, noncurrent_version_transition.value.noncurrent_days, null)
+          storage_class             = noncurrent_version_transition.value.storage_class
+        }
+      }
+
+      dynamic "transition" {
+        for_each = try(flatten([rule.value.transition]), [])
+        content {
+          date          = try(transition.value.days, null) != null ? null : try(transition.value.date, null)
+          days          = try(transition.value.days, 0)
+          storage_class = transition.value.storage_class
+        }
+      }
+    }
+  }
+
+  # Must have bucket versioning enabled first
+  depends_on = [aws_s3_bucket_versioning.main]
```

Comment:
And there is no mention on bucket versioning on features

Reply:
Done